utils.go

```diff
@@ -290,7 +291,29 @@ func (s *Storage) formatFileObject(v *s3.Object) (o *typ.Object, err error) {
 	o = s.newObject(false)
 	o.ID = *v.Key
 	o.Path = s.getRelPath(*v.Key)
-	o.Mode |= typ.ModeRead
+
+	headInput := &s3.HeadObjectInput{
+		Bucket: aws.String(s.name),
+		Key:    aws.String(*v.Key),
+	}
+
+	headOutput, err := s.service.HeadObject(headInput)
```

Comment:
`HeadObject` here will add pressure for `List`?

Reply:
I think it will increase the pressure, but I haven't found a better way to implement it.

Reply:
Add `HeadObject` in `ListObject` will make it much slower. As we are implementing a virtual link, maybe it's possible to allow not reflected during the list?

Let's discuss it in the RFC instead.